parsers/csv/csv.go

```diff
@@ -0,0 +1,152 @@
+package csv
+
+import (
+	"encoding/csv"
+	"errors"
+	"io"
+	"math"
+	"strings"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+
+	"github.com/honeycombio/honeytail/event"
+	"github.com/honeycombio/honeytail/httime"
+	"github.com/honeycombio/honeytail/parsers"
+)
+
+// Options defines the options relevant to the CSV parser
+type Options struct {
+	Fields          string `long:"fields" description:"Comma separated list of CSV fields, in order."`
+	TimeFieldName   string `long:"timefield" description:"Name of the field that contains a timestamp"`
+	TimeFieldFormat string `long:"time_format" description:"Timestamp format to use (strftime and Golang time.Parse supported)"`
+	NumParsers      int    `hidden:"true" description:"number of csv parsers to spin up"`
+}
+
+// Parser implements the Parser interface
+type Parser struct {
+	conf       Options
+	lineParser parsers.LineParser
+}
+
+// Init constructs our parser from the provided options
+func (p *Parser) Init(options interface{}) error {
+	p.conf = *options.(*Options)
+	if p.conf.Fields == "" {
+		return errors.New("must provide at least 1 field name when parsing CSV lines")
+	}
+	lineParser, err := NewCSVLineParser(p.conf.Fields)
+	if err != nil {
+		return err
+	}
+	p.lineParser = lineParser
+	return nil
+}
+
+type CSVLineParser struct {
+	fields []string
+}
+
+// NewCSVLineParser factory
+func NewCSVLineParser(fieldsString string) (*CSVLineParser, error) {
+	// Is building a reader for every single line a good idea?
+	// Potential for future optimization here
+	reader := strings.NewReader(fieldsString)
+	csvReader := csv.NewReader(reader)
+	csvReader.TrimLeadingSpace = true
```

Comment:
We set `TrimLeadingSpace = true` here, but not on the `csvReader` down in `ParseLine()` -- probably best to be consistent.

Reply:
I'm going to make this optional for both fields and values, defaulting to false.